src/main/java/me/botsko/prism/actions/BlockAction.java

```diff
@@ -30,574 +30,534 @@
-			stateChange = new BlockStateChange(originalBlock, originalBlock);
-
-			// Preview it
-			EntityUtils.sendBlockChange(player, block.getLocation(), getBlockData());
-
-			// Send preview to shared players
-			for (final CommandSender sharedPlayer : parameters.getSharedPlayers()) {
-				if (sharedPlayer instanceof Player) {
-					EntityUtils.sendBlockChange((Player) sharedPlayer, block.getLocation(), getBlockData());
-				}
-			}
-		}
-
-		return new ChangeResult(ChangeResultType.APPLIED, stateChange);
-
-	}
-
-	/**
-	 * 
-	 */
-	protected ChangeResult removeBlock(Player player, QueryParameters parameters, boolean is_preview, Block block) {
-
-		BlockStateChange stateChange;
-
-		if (!block.getType().equals(Material.AIR)) {
-
-			// Ensure it's acceptable to remove the current block
-			if (!BlockUtils.isAcceptableForBlockPlace(block.getType())
-					&& !BlockUtils.areBlockIdsSameCoreItem(block.getType(), getMaterial())
-					&& !parameters.hasFlag(Flag.OVERWRITE)) {
-				return new ChangeResult(ChangeResultType.SKIPPED, null);
-			}
-
-			if (!is_preview) {
-
-				// Capture the block before we change it
-				final BlockState originalBlock = block.getState();
-
-				// Set
-				block.setType(Material.AIR);
-
-				// Capture the new state
-				final BlockState newBlock = block.getState();
-
-				// Store the state change
-				stateChange = new BlockStateChange(originalBlock, newBlock);
-
-			}
-			else {
-
-				// Otherwise, save the state so we can cancel if needed
-				final BlockState originalBlock = block.getState();
-				// Note: we save the original state as both old/new so we can
-				// re-use blockStateChanges
-				stateChange = new BlockStateChange(originalBlock, originalBlock);
-
-				// Preview it
-				EntityUtils.sendBlockChange(player, block.getLocation(), Bukkit.createBlockData(Material.AIR));
-
-				// Send preview to shared players
-				for (final CommandSender sharedPlayer : parameters.getSharedPlayers()) {
-					if (sharedPlayer instanceof Player) {
-						EntityUtils.sendBlockChange((Player) sharedPlayer, block.getLocation(),
-								Bukkit.createBlockData(Material.AIR));
-					}
-				}
-			}
-			return new ChangeResult(ChangeResultType.APPLIED, stateChange);
-		}
-		return new ChangeResult(ChangeResultType.SKIPPED, null);
-	}
+    private BlockActionData actionData;
+
+    /**
+     * Set the Block.
+     *
+     * @param block Block
+     */
+    public void setBlock(Block block) {
+        if (block != null) {
+            setBlock(block.getState());
+        }
+    }
+
+    /**
+     * Set the Block State.
+     *
+     * @param state BlockState.
+     */
+    public void setBlock(BlockState state) {
+        if (state != null) {
+            setMaterial(state.getType());
+            setBlockData(state.getBlockData());
+            createActionData(state);
+            setLoc(state.getLocation());
+        }
+    }
+
+    private void createActionData(BlockState state) {
+        switch (state.getType()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A switch statement does not contain a break](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5446892)